### ISSUE TITLE
Invalid snippets

### ISSUE THREAD

**Issue by sergeche** [emmetio/sublime-text-plugin#100]
Hi @sergeche !
Please help me to deal with my CSS/HTML snippets.

Before Emmet update:
- "bd1" => "border:1px solid #|;"; 
- "c1" => "color:#111;"; 
- "cf" => "color:#fff;"; 
- "lh20" => "line-height:20px;"; 
- "after" => "display:block;\ncontent:\"\";\nclear:both;" 

After Emmet update:
- "bd1" => "border: 1px;"; 
- "c1" => "color: 1px;"; 
- "cf" => NOTHING; 
- "lh20" => "line-height: 20;"; 
- "after" => NOTHING

... and a lot of such examples I've got. Also I can't find how to manage css.valueSeparator option. Previously it was "css.valueSeparator": ":" (without spaces after ":"), but now it doesn't work as well. Another question is how to create HTML snippet with caret position after abbreviation expanding? Sequence of caret position or highlight some code after expanding? For example:
`"js": "<script src=\"js/${1:main.js}\">${2}</script>${3}"`

My old snippets and preferences:
[emmet-st3.zip](https://github.com/emmetio/sublime-text-plugin/files/5152535/emmet-st3.zip)

Thank you in advance :)

_Originally posted by @gladisihor in https://github.com/emmetio/sublime-text-plugin/issues/50#issuecomment-684024126_

**Comment by sergeche** [comment]
@gladisihor Emmet config format has been changed: https://github.com/emmetio/sublime-text-plugin#adding-custom-emmet-snippets

Also, I see you use Emmet in a bit strange way. The main goal of Emmet is to provide a basis for CSS properties, which you can extend in real-time by adding values right into abbreviation. Instead, you’ve created lots of snippets with predefined values like `fz11` which is actually resolves to `font-size: 11px;` without any additional snippets.

So, if you use Emmet _that way_, your best option is to not use Emmet at all and use default ST snippets/completions instead

**Comment by gladisihor** [comment]
Спасибо за Ваш ответ! 
Здесь уже скорее вопрос привычки. Подобные сниппеты очень сильно экономят моё время и нарабатывались не один год. Очень хотелось бы пользоваться ими и в дальнейшем. К тому же дефолтные сниппеты ST вряд ли откроются без ST хинтов и по нажатию на привычную для меня комбинацию клавишь (у меня за это отвечает сочетание ALT+SPACE). Может всё-таки есть какой-то способ вернуть их к жизни? Или возможно, есть способ сделать downgrade для Emmet?

**Comment by sergeche** [comment]
Да, можно: я чуть позже доработаю Emmet, чтобы он в том числе резолвил сниппеты типа `fz11`, то есть чтобы принудительно не отделял значения. Выше указал, как добавлять Emmet-сниппеты в ST, там же есть ссылка на параметр для разделителя (сейчас это `"stylesheet.between"`).

А по поводу финального таб-стопа — да, про это забыл, уже есть такой баг: #97

**Comment by Eugene-I** [comment]
> 
> 
> Да, можно: я чуть позже доработаю Emmet, чтобы он в том числе резолвил сниппеты типа `fz11`, то есть чтобы принудительно не отделял значения. Выше указал, как добавлять Emmet-сниппеты в ST, там же есть ссылка на параметр для разделителя (сейчас это `"stylesheet.between"`).
> 
> А по поводу финального таб-стопа — да, про это забыл, уже есть такой баг: #97

я уже попробовал добавить свой снипет в настройки, но по умолчанию все равно срабатывает дефолтный. Например при m3  у меня разворачивается margin:3px;, хотя для себя я задавал margin:0 0 10px;
если задать абревиатуру, которая не конфликтует с дефолтной, например marg3, то тогда все работает как надо.
У всех просто кастомные снипеты забиты в джейсоне.
Может все-таки лучше откатить пока новый функционал хотя бы для саблайма? Реально как без рук сидишь. К тому же это затронуло и остальные синтаксисы. В том же ПАГе не работает теперь нормально автокомплит, когда после А нажимаешь таб и в результате получаешь a(href="#")

**Comment by sergeche** [comment]
А для чего вы в PUG после `a` нажимаете Tab? Подсказка: Esc сбросит захваченную аббревиатуру и дальше Tab можно использовать как раньше, вставлять таб или разворачивать нативный сниппет

**Comment by Eugene-I** [comment]
> 
> 
> А для чего вы в PUG после `a` нажимаете Tab? Подсказка: Esc сбросит захваченную аббревиатуру и дальше Tab можно использовать как раньше, вставлять таб или разворачивать нативный сниппет

что бы добавились атрибуты по умолчанию. С img это тоже не работает - там атрибутов побольше. Не разворачивает вообще)

**Comment by sergeche** [comment]
Я не понимаю, в чём именно проблема? Можете подробнее написать, что делаете, что получаете и что ожидаете?

**Comment by Eugene-I** [comment]
`{
	"config": {
		"scss": {
			"snippets": {
				"m3": "margin:0 0 10px;"
			}
		}
	}
}`

Это попытка настроить эммет так как мне надо.
Но при попытке воспользоваться этим снипетом мне предлагает дефолтный вариант https://take.ms/eoHdn

**Comment by sergeche** [comment]
Я имел в виду проблему с PUG

**Comment by Eugene-I** [comment]
lorem не генерирует
https://take.ms/D1TBr
при попытке использовать img снипет ничего не подставляет. Хотя в версии 1 после нажатия на таб было развертывание до `img(src="images/", alt="Image description", width="", height="")`

https://take.ms/vdbCo

**Comment by sergeche** [comment]
В настройках Emmet в секцию `abbreviation_scopes` добавьте `"text.pug - entity - meta"`. Он там просто не работает сейчас

**Comment by Eugene-I** [comment]
без изменений
вот скрин настроек https://take.ms/3s00E

**Comment by sergeche** [comment]
Что именно не работает? Сниппет `img` или Emmet вообще (даже `a` не разворачивает)? Какой синтаксис для PUG используете? И в `abbreviation_scopes` надо добавлять к имеющимся новое значение, вы перетёрли скоупы и теперь нигде кроме PUG работать не будет

**Comment by Eugene-I** [comment]
PUG можно пока опустить. Он там вообще не работает. Насчет синтаксиса не понял :) можно просто новый файл index.pug создать и попробовать создать любой тег - ничего не выйдет. Только стандартный автокомплит.
В scss не работают те сниппеты, которые я добавляю в настройки.
Мне кажется что если Вам установить саблайм, установить эммет и попробовать его кастомизировать - то тоже ничего не получится

**Comment by sergeche** [comment]
Про сниппеты типа `fz11` и `m3`, где числовое значение зашито в сам спиппет я уже написал — нужно доработать Emmet, чтобы такое поддерживалось (грубо говоря, это не совсем корректное использование Emmet и для таких целей лучше использовать нативные сниппеты, но хрен с ним, сделаю). Мы сейчас решаем проблему с PUG.

> Насчет синтаксиса не понял :)

В Sublime Text документы красятся через специальные синтаксисы, которые распространяются в виде пакетов. Интересует, какой пакет используется для синтаксиса Pug. Я проверил только что предложенное мной решение, всё работает. Хочу понять, что не работает у вас

**Comment by Eugene-I** [comment]
https://packagecontrol.io/packages/Pug
Этот. А Вы?)

**Comment by sergeche** [comment]
Этот же. С ним и указанной опцией всё работает как надо
